hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveStatelessSessionImpl.java

```diff
@@ -150,10 +150,12 @@ private ReactiveStatelessSessionImpl(
 			PersistenceContext persistenceContext) {
 		super( factory, options );
 		this.persistenceContext = persistenceContext;
-		Integer batchSize = getConfiguredJdbcBatchSize();
-		reactiveConnection = batchSize == null || batchSize < 2
-				? connection
-				: new BatchingConnection( connection, batchSize );
+		// Don't know why ORM set the batch size to 0 in the super constructor,
+		// but the default should be the one in the factory options
+		int defaultBatchSize = factory.getSessionFactoryOptions().getJdbcBatchSize();
+		setJdbcBatchSize( defaultBatchSize );
+		// BatchingConnection will use the original connection if defaultBatchSize <= 1
+		reactiveConnection = new BatchingConnection( connection, defaultBatchSize );
```

Comment:
Need to be careful about this. I'm not certain, but I think this might have the side-effect of making the stateless session asynchronous, which we don't want it to be by default.

Reply:
Yeah, so I just checked. Reconstructing the history behind this, it looks like historically `StatelessSession` did not allow JDBC batching, because that would change it's "immediate synchronous execution" model into something more like transactional write-behind and be confusing.

It appears that Vlad _accidentally_ added `setJdbcBatchSize()` to `StatelessSession`, breaking its historical guarantees, in https://hibernate.atlassian.net/browse/HHH-10431. I've attempted to move _back away from that_ in H7 by introducing `insertMultiple()` and friends as an alternative.

Anyway:

1. I would prefer if `StatelessSession` did _not_ have a way to set a session-level batch size, and
2. it certainly should default to a batch size of zero.

So I think this change is wrong.

Reply:
OK, it makes sense now. Thanks.

> 1. I would prefer if StatelessSession did not have a way to set a session-level batch size, and

I don't think we need them in the API, everything should work if we change the type of `batchingHelperSession`  to `ReactiveStatelessSessionImpl` (in `ReactiveStatelessSessionImpl`)

> 2. it certainly should default to a batch size of zero.

I think we can just change the original code to
```java
```

Reply:
OK, it makes sense now. Thanks.

> 1. I would prefer if StatelessSession did not have a way to set a session-level batch size, and

I don't think we need them in the API, everything should work if we change the type of `batchingHelperSession`  to `ReactiveStatelessSessionImpl` (in `ReactiveStatelessSessionImpl`)

> 2. it certainly should default to a batch size of zero.

I think this means that we can now simplify the code to 
```java
reactiveConnection = new BatchingConnection( connection, 0 );
```
but, `statelessSession.insertAll(objects)` will never use batching, right?.

Reply:
if we cannot also set the jdbc bacth size of the batchingHelperSession then methods like `insertAll(int batchSize, ... )` are not going to work  because we are going to use `new BatchingConnection( connection, 0 )` that will not execute any batch.

Reply:
@DavideD 

`statelessSession.insertAll(objects)` will use batching because it's directly calling `setJdbcBatchSize()` on the internal SPI. And since it uses _exactly the right batch size_, there's no write-behind behavior.

Reply:
so, insertAll will always batch, right?

Reply:
That's the idea, yeah.

Reply:
working on it

Reply:
@gavinking pushed the changes.